packages/core/src/shared/clients/docdbClient.ts

```diff
@@ -64,6 +64,8 @@ export class DefaultDocumentDBClient {
             return response.DBEngineVersions ?? []
         } catch (e) {
             throw ToolkitError.chain(e, 'Failed to get DocumentDB engine versions')
+        } finally {
+            client.destroy()
```

Comment:
Suggestion, perhaps the client would be a good use case for the `using` feature in JS.

https://www.typescriptlang.org/docs/handbook/release-notes/typescript-5-2.html

Reply:
True, unfortunately this project is using an older version.

Reply:
Oh, wow. I thought 5.2 was pretty old. Carry on.